src/odemis/driver/andorshrk.py

```diff
@@ -777,14 +850,15 @@ def SetWavelength(self, wavelength):
                     self._dll.ShamrockSetWavelength(self._device, c_float(wavelength * 1e9))
                 except ShamrockError as ex:
                     if ex.errno != 20201 or retry >= 5: # SHAMROCK_COMMUNICATION_ERROR
-                        raise
+                        self._reconnect()
```

Comment:
Only if errno is  in `(20201, 20275)`, reconnect, otherwise raise.

Reply:
I'm not sure if this is going to work. I think it sometimes also raises the 20201 error that has nothing to do with reconnect.